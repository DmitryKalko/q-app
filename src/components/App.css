@import url("https://fonts.googleapis.com/css2?family=Spartan&display=swap");

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: "Spartan", sans-serif;
}
.main {
  background: rgb(134, 62, 70);
  background: linear-gradient(
    0deg,
    rgb(134, 62, 70) 0%,
    rgba(249, 146, 156, 1) 100%
  );
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}
.form {
  width: 80%;
  max-width: 1000px;
  height: 80vh;
  max-height: 700px;
  display: flex;
  border-radius: 2%;
  -webkit-box-shadow: 0px 27px 90px 42px rgba(0, 0, 0, 0.2);
  -moz-box-shadow: 0px 27px 90px 42px rgba(0, 0, 0, 0.2);
  box-shadow: 0px 27px 90px 42px rgba(0, 0, 0, 0.2);
}

/* switcher mode */
.changeTheme {
  align-self: flex-end;
  display: flex;
  position: absolute;
  top: 10px;
  right: 0;
}
/* --- */

.left-part {
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgb(80, 194, 183);
  background: linear-gradient(
    40deg,
    rgb(80, 194, 183) 0%,
    rgba(106, 219, 199, 1) 100%
  );
  width: 50%;
  max-width: 500px;
  height: 100%;
  border-radius: 2% 0 0 2%;
}
.right-part {
  position: relative;
  background: rgb(33, 46, 63);
  background: linear-gradient(
    40deg,
    rgba(33, 46, 63, 1) 0%,
    rgba(56, 75, 98, 1) 100%
  );
  width: 50%;
  max-width: 500px;
  height: 100%;
  border-radius: 0 2% 2% 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
.step {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 1.8em;
  padding: 40px;
  text-align: center;
}
.step img {
  width: 100%;
  height: auto;
  max-height: 450px;
}
.component {
  width: 100%;
  height: 80%;
  display: flex;
  justify-content: center;
  align-items: center;
}
.first-step,
.second-step {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 80%;
}
.first-step input {
  width: 90%;
  height: 50px;
  margin: 20px 0;
  padding-left: 30px;
  outline: none;
  background-color: rgba(0, 0, 0, 0.397);
  color: white;
  border-radius: 10px;
  border: 0;
  font-size: 1em;
}
.btn {
  background: rgb(80, 194, 183);
  background: linear-gradient(
    40deg,
    rgb(80, 194, 183) 0%,
    rgba(106, 219, 199, 1) 100%
  );
  border: 0;
  border-radius: 10px;
  width: 30%;
  height: 50px;
  color: white;
  font-size: 1em;
  cursor: pointer;
}
.btns-block {
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  margin-top: 20px;
}

.Mui-checked {
  color: #67d8c5 !important;
}

@media (max-width: 600px) {
  .step p {
    font-size: 17px;
  }
  .btn {
    font-size: 14px;
    width: 40%;
  }
  .first-step input {
    font-size: 12px;
    padding-left: 3px;
  }
  .second-step,
  .third-step {
    width: 90%;
  }
}
