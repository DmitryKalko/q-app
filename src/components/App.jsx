import React, { useState, useEffect } from "react";

import FirstStep from "./FirstStep";
import SecondStep from "./SecondStep";
import ThirdStep from "./ThirdStep";
import FirstStepDescription from "./FirstStepDescription";
import SecondStepDescription from "./SecondStepDescription";
import ThirdStepDescription from "./ThirdStepDescription";
import CustomizedSwitches from "./ThemeSwitcher";

import "./App.css";

import data from "../store/data";

const App = () => {
  const [step, setStep] = useState(1);
  const [lightTheme, setLightTheme] = useState(true);
  const [stateData, setStateData] = useState({});

  useEffect(() => {
    setStateData(data.store);
  }, []);

  const switchTheme = () => {
    setLightTheme(!lightTheme);
  };

  const toSecondStep = () => {
    if (stateData.firstName && stateData.lastName) {
      setStep(step + 1);
    } else {
      validationAlert();
    }
  };

  const toThirdStep = () => {
    if (stateData.gender && stateData.policyAgree) {
      setStep(step + 1);
    } else {
      validationAlert();
    }
  };

  const toPrevStep = () => {
    setStep(step - 1);
  };

  const validationAlert = () => {
    alert("Please fill in all fields");
  };

  const showAlert = () => {
    if (stateData.notifications) {
      alert("You have successfully completed all the steps!");
      alert(JSON.stringify(data));
      location.reload();
    } else {
      validationAlert();
    }
  };

  return (
    <div className="main">
      <div className="form">
        <div className="left-part">
          {step === 1 && <FirstStepDescription lightTheme={lightTheme} />}
          {step === 2 && <SecondStepDescription lightTheme={lightTheme} />}
          {step === 3 && <ThirdStepDescription lightTheme={lightTheme} />}
        </div>

        <div className="right-part">
          <div className="changeTheme">
            <CustomizedSwitches switchTheme={switchTheme} />
          </div>
          <div className="component">
            {step === 1 && (
              <FirstStep toNextStep={toSecondStep} lightTheme={lightTheme} />
            )}
            {step === 2 && (
              <SecondStep
                toNextStep={toThirdStep}
                toPrevStep={toPrevStep}
                lightTheme={lightTheme}
              />
            )}
            {step === 3 && (
              <ThirdStep
                toFinishStep={showAlert}
                toPrevStep={toPrevStep}
                lightTheme={lightTheme}
              />
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default App;
