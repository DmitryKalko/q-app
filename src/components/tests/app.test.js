import { shallow } from "enzyme";
import {jest} from '@jest/globals'
import React from "react";
import App from "../App";


const setUp = (props) => shallow(<App {...props}/>);  // shallow осуществляет неглубокую отрисовку - рендерится только наш компенент, без дочерних

describe('should render App component', () => {
    let component;
    beforeEach(() => {   // будет выполнять логику перед запуском каждого из тестов. полезно бля однотипных тестов
        component = setUp();
    });
    it('should contain .main wrapper', () => {
        const wrapper = component.find('.main');  // находим класс main. можно искать и по тегам
        expect(wrapper.length).toBe(1); // проверям что он встречается только один раз в нашем компоненте
        // console.log(component.debug());
    });
    it('should contain .main wrapper', () => {
        const wrapper = component.find('div'); 
        expect(wrapper.length).toBeGreaterThan(1); 
        // console.log(component.debug());
    });
});


describe('make snaphots', () => {
    it('should shallow App component', () => {
        const component = shallow(<App/>)   // shallow осуществляет неглубокую отрисовку - рендерится только наш компенент, без дочерних. по всемени - самый быстрый
        expect(component).toMatchSnapshot();
    });
    // it('should render App component', () => {
    //     const component = render(<App/>)  // рендерится целевой компонент с дочерними. используется именно для проверки разметки. по времени - средний
    //     expect(component).toMatchSnapshot();
    // });
    // it('should mount  App component', () => {
    //     const component = mount(<App/>)     // рендерится целевой компонент с дочерними. используется если нужно тестить методы жизненного цикла или поведение дочерних элементов. по времени - самый долгий
    //     expect(component).toMatchSnapshot();
    // });

    it('should handle the click event', () => {
        window.alert = jest.fn();                     // мокаем функцию alert
        const component = shallow(<App/>)
        component.simulate('click');                 // эмитируем клик
        expect(window.alert).toHaveBeenCalledWith('Please fill in all fields');
      });
});

it('should do some stuff on window resize event', () => {
    const wrapper = mount(<App />);
    const spy = jest.spyOn(wrapper.instance(), 'handleWindowResize');

    // set spy to 'resize' listener
    global.addEventListener('resize', spy);
    global.dispatchEvent(new Event('resize'));

    expect(spy).toHaveBeenCalled();
});


