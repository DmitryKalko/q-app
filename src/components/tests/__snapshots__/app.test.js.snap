// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snaphots should shallow App component 1`] = `
<div
  className="main"
>
  <div
    className="form"
  >
    <div
      className="left-part"
    >
      <FirstStepDescription
        lightTheme={true}
      />
    </div>
    <div
      className="right-part"
    >
      <div
        className="changeTheme"
      >
        <CustomizedSwitches
          switchTheme={[Function]}
        />
      </div>
      <div
        className="component"
      >
        <Memo(observerComponent)
          lightTheme={true}
          toNextStep={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
